### PR TITLE
feat: Enable snyk monitoring

### PR DESCRIPTION
**Pull request by roseteague**
Part of osp-cfc-platform/backlog#1045

### CODE REVIEW

**Comment by roseteague** [comment]
node_js version >=6 required in order to install snyk

Refactoring travis-ci to improve consistency across repos

File: .travis/docker-build.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+testSnykIfEnabled() {
+  if [ -n "${SNYK_ORG}" ] && [ -n "${SNYK_TOKEN}" ]; then
+      local errors_found=false
+      snyk --version
```
Comment (rubencabrera) [inline .travis/docker-build.sh]: Why is this needed?

File: .travis/docker-push.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+performSnykAnalysisIfEnabled() {
+  if [ -n "${SNYK_ORG}" ] && [ -n "${SNYK_TOKEN}" ]; then
+      snyk monitor --org="${SNYK_ORG}" --docker "${TRAVIS_REPO_SLUG}:${TRAVIS_COMMIT}" --policy-path=.snyk
```
Comment (rubencabrera) [inline .travis/docker-push.sh:5]: There is no `.snyk` file in this repo yet. I know we've already done the same thing in other repos and it works, but it just occurred to me that it might be a bit confusing if someone tries to find it. Should we add an empty one or a comment before this line?

Should we add the dotfile and put the

File: .travis.yml [hunk 3]
```diff
@@ -1,44 +1,30 @@
...
+  - secure: "meptwbsQls6iwpPaDT4z7wrgdeLt95iS/q/MY7L52YEua+7tknhCLOknZypZ9OzjpYb0CPQm6eRzG+g8TLNI3e0lKdR3skp7aumhDUPRhuXusYADrr14E7qBsfm3Kaahb3MaMKiKfJox0k2DBvEjlBq2zY+kHnJCj3BjzCte72ezqTu1G6yTHTXvuhmHwVkkpU5JjoT6UWF6l4Ot7AerBb970jAb/WK73l28vhhHzPTljLuYzdf+fI1N8f2DdPYhY0MeFeu85VNftWpmv4g7dsxlkOqPZDEWdJVEDO0zYQmpSlFGRZquRTN3YUgH8Rdezr5AFUOEE5ayXHRY7iVI1cT5e65mSkRYXhJMH47VdZhNtZ6oQ12OKP9cEEkd/ccwMh3Q0mAybuwElCxnucAMnYT5CT6+KNULWeVVTfOZmcJZbfw+3lsCqZ7AxGruEhMfrVDgH2Kk9BvF3yt0mABxqZR0u1+QqTiaaJcUi4akVa8C6kf4n5sflZMU5O+LLqE65trGi9YvId9Ir7opqZtDIkp/xdbi84/icVnO1AH9iGToGE+7mD3/kPbmqrnhRqjAC6WpwptSTmm7iYrw11De/tMx3A1rhH/+ohlaBiGCt3VR1rcBLcxr0rIR/El93wAqqn9VZgsDJ/BiWzQVJgt2QnU9frsa88K2i4vdp4MQQ0o="
```
Comment (stuart-warren) [inline .travis.yml]: what is in these 3 secrets? not the actual values obviously

**Review by stuart-warren — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 3 above)
Comment (roseteague) [reply .travis.yml]: I think the top one is the registry pass, and the other two are the snyk org and snyk token

File: .travis/docker-build.sh (hunk 1 above)
Comment (roseteague) [reply .travis/docker-build.sh]: It's not, I just forgot to delete it, thanks!

File: .travis/docker-push.sh (hunk 2 above)
Comment (roseteague) [reply .travis/docker-push.sh:5]: That's a good point. I could add a comment as that will probably be easiest to understand

File: .travis.yml (hunk 3 above)
Comment (stuart-warren) [reply .travis.yml]: Hmmm, maybe the secrets could/should be commented so we know which is which?

File: .travis/docker-push.sh (hunk 2 above)
Comment (roseteague) [reply .travis/docker-push.sh:5]: ^^ See added section to the README

File: .travis.yml (hunk 3 above)
Comment (roseteague) [reply .travis.yml]: will do that now

Comment (roseteague) [reply .travis.yml]: done

Comment (stuart-warren) [reply .travis.yml]: thanks, sorry, would it be possible for the comments to match the environment vars they are?

Comment (roseteague) [reply .travis.yml]: eg `#SNYK_ORG=` ?

Comment (stuart-warren) [reply .travis.yml]: exactly

Comment (roseteague) [reply .travis.yml]: That should be right now?

**Review by stuart-warren — APPROVED** [review]

File: .travis/docker-push.sh (hunk 2 above)
Comment (rubencabrera) [reply .travis/docker-push.sh:5]: Perfect, the README is the best place.
